src/database.py

```diff
@@ -445,3 +445,12 @@ def splitTokens(candidates: str) -> (str, str, str):
                 continue
             data.append({"key": key, "operation": operation, "value": converted_value})
         return data
+
+    def getBGGIDFromName(self, name: str) -> [int]:
+        cursor = self.connection.cursor()
+        # remove case sensitivity and return all results that start with 'name' str. BGG can handle up to 20 per request.
+        data = cursor.execute(
+            f"SELECT boardgames.bgg_id FROM items JOIN boardgames ON items.id = boardgames.id WHERE items.name LIKE {name}% COLLATE NOCASE"
+        ).fetchall()
+        ids = [item['bgg_id'] for item in data]
+        return ids if data is not None else []
```

Comment:
If data can be None, won't the list comprehension right above throw and exception in that case?

Reply:
This is a gap in my knowledge of python. Should `return ids if len(data) > 0 else []` be better?

Reply:
None is equivalent to NULL. 
I think the simplest will be to just put `return ids` since if data is empty then ids will be too, so ids will be equal to []

Reply:
aight. fixing it now